og.module

```diff
@@ -224,30 +225,30 @@ function _og_modules_uninstalled($modules) {
 /**
  * Implements hook_og_permission().
  */
-function _og_og_permission() {
-  // Generate standard node permissions for all applicable node types.
-  $perms = array();
+function og_og_permission() {
+  $permissions = [
+    'update group' => [
+      'title' => t('Edit group'),
+      'description' => t('Edit the group. Note: This permission controls only node entity type groups.'),
+      'default role' => array(OgRoleInterface::ADMINISTRATOR),
+    ],
+    'administer group' => [
+      'title' => t('Administer group'),
+      'description' => t('Manage group members and content in the group.'),
+      'default role' => array(OgRoleInterface::ADMINISTRATOR),
+      'restrict access' => TRUE,
+    ],
+  ];
 
-  $perms['update group'] = array(
-    'title' => t('Edit group'),
-    'description' => t('Edit the group. Note: This permission controls only node entity type groups.'),
-    'default role' => array(OgRoleInterface::ADMINISTRATOR),
-  );
-  $perms['administer group'] = array(
-    'title' => t('Administer group'),
-    'description' => t('Manage group members and content in the group.'),
-    'default role' => array(OgRoleInterface::ADMINISTRATOR),
-    'restrict access' => TRUE,
-  );
-
-  foreach (node_permissions_get_configured_types() as $type) {
-    $perms = array_merge($perms, og_list_permissions($type));
+  foreach (Og::groupManager()->getAllGroupBundles() as $entity_type_id => $bundle_ids) {
+    foreach ($bundle_ids as $bundle_id) {
+      $permissions = array_merge($permissions, OgPermissionHelper::generatePermissionList($entity_type_id, $bundle_id));
+    }
```

Comment:
I ported this directly from D7, but there seems to be a problem in the original approach. In the original D7 code `og_list_permissions()` will check for every node type if it is a group content type, and if it is the case permissions will be added to create, edit and delete the group content.

But this doesn't take into account that group content can be limited per entity type and bundle. It is possible that a certain group content is only targeting the 'group1' bundle and not the 'group2' bundle. Meaning that the permissions will be different for each group bundle.

Don't we need to update this hook to pass in the target group and target bundle? Otherwise the permissions that are generated might or might not be applicable to a certain group type, rendering it effectively useless for generating permission tables.

So in short, shouldn't this be:

```
/**
 * Return a list of permissions for the given group type.
 */
function hook_og_permission($group_type, $group_bundle) {
}
```